src/commands/test.js

```diff
@@ -214,7 +224,7 @@ async function runDocker(datasource, opts) {
     // else it'll return the image ID. Skip `docker build` if an image already exists
     // If `-v/--version` is specified, delete current image(if any) and rebuild.
     // Use spawn() and {stdio: 'inherit'} so we can see the logs in real time.
-    if(stdout === '' || versionOpt || forceOpt) {
+    if(!dockerfileExists || stdout === '' || versionOpt || forceOpt) {
```

Comment:
Why if the `Dockerfile` exists it shouldn't remove the image?

Reply:
Here the idea is that it should build the image if the Dockerfile did not exists at the time the `graph test -d` command was run, an image does not exists, or the rebuild is forced by passing a matchstick version or the force tag.